tests/unit/parsers/test.manifestjson.js

```diff
@@ -3601,4 +3601,125 @@ describe('ManifestJSONParser', () => {
       );
     });
   });
+
+  describe('install_origins', () => {
+    function testInstallOrigins(origins, expectValid) {
+      const addonLinter = new Linter({ _: ['bar'] });
+      const json = validManifestJSON({
+        browser_specific_settings: {
+          gecko: {
+            id: 'foo@bar',
+            install_origins: origins,
+          },
+        },
+      });
+      const manifestJSONParser = new ManifestJSONParser(
+        json,
+        addonLinter.collector
+      );
+      expect(manifestJSONParser.isValid).toEqual(expectValid);
+      return addonLinter.collector;
+    }
+
+    it('allows install origins with valid origin', () => {
+      testInstallOrigins(['https://example.com'], true);
+    });
+
+    it('allows install origins with subdomain origin', () => {
+      testInstallOrigins(['https://foo.example.com'], true);
+    });
+
+    it('allows install origins with punnycode origin', () => {
+      testInstallOrigins(['https://xn--fo-9ja.com'], true);
+    });
+
+    it('allows install origins with IDN origin', () => {
+      testInstallOrigins(['https://foo.bar.栃木.jp'], true);
+    });
+
+    it('allows install origins with port', () => {
+      testInstallOrigins(['https://example.com:8888'], true);
+    });
```

Comment:
Nit, we may use `it.each` Jest helper as we do for some other similar test cases, e.g. this one related to the homepage_url:
https://github.com/mozilla/addons-linter/blob/563ad4ffa2cb877fbc94c29debf5a3759aad18a9/tests/unit/parsers/test.manifestjson.js#L2977-L2982

Reply:
nit: also if we can avoid `testXXX` and have the assertions directly in the test function, that would avoid some ESLint warnings about tests without assertions :)